### PR TITLE
[java] [okhttp-gson] Update OkHttp to version 3.x

### LINKED ISSUE #9606: [Java] [okhttp-gson] Updating OkHttp to version 3.x
##### Description

I'd like to gauge the appetite for upgrading the version of OkHttp in the `okhttp-gson` module to a version in the 3.x series. Version 2.7.5 dates back to Feb 26, 2016, and there hasn't been an update since then.

Happy to put together a patch just for this module, but I first wanted to check to see if this is something that makes sense? Would this type of change need to wait for a major release of swagger-codegen, or could it be something that's released in a minor version - from what I can tell there would be API changes given the OkHttp package name is different in version 3.x and the API in okhttp-gson exposes some OkHttp classes.

##### Swagger-codegen version

3.0.10.

##### Related issues/PRs

This would help make progress on kubernetes-client/java#430. We'd love to be using a more recent version of OkHttp!

##### Suggest a fix/enhancement

Would be willing to offer up a patch (I have something already) if there's someone that would be willing to guide me through the process (my first time committing to this repo, so I might be missing some subtleties).

### PR DESCRIPTION
**Pull request by nicktrav**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

The current implementation of the okhttp-gson client uses OkHttp 2.x as
the underlying client. This library was superseded by version 3.x in
early 2016. Version 3.x of the library is still under active
development.

Update codegen templates for okhttp-gson to use version 3.x of the
OkHttp client library.

Closes #9606.

### CODE REVIEW

**Comment by nicktrav** [comment]
cc: @frantuma

**Comment by nicktrav** [comment]
This is a first pass at porting to OkHttp 3.x.

Please let me know if it makes sense to have this in a completely separate package. The changes aren't too invasive (only one new public method was added), but as mentioned on #9606, this is technically a breaking change as the underlying okhttp package name changed to `com.squareup.okhttp3`, which changes the signatures on the APIs.

There were some seemingly unrelated changes to some files after running the `./bin/security/java-petstore-okhttp-gson.sh` (probably due to changes not being checked in in the past).

Unit tests in `samples/client/petstore/java/okhttp-gson` and `samples/client/petstore-security-test/java/okhttp-gson` compile and pass. Not sure if there are any other tests that need updating.

File: modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache [hunk 1]
```diff
@@ -2,10 +2,20 @@
 
 package {{invokerPackage}};
 
-import com.squareup.okhttp.*;
```
Comment (nicktrav) [inline modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache]: The Google styleguide discourages wildcards:
https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports 

If it's preferable to use wildcards from a codegen perspective, I can revert.

File: modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache [hunk 2]
```diff
@@ -454,7 +468,9 @@ public class ApiClient {
      * @return Api client
      */
     public ApiClient setConnectTimeout(int connectionTimeout) {
-        httpClient.setConnectTimeout(connectionTimeout, TimeUnit.MILLISECONDS);
+        httpClient = httpClient.newBuilder()
```
Comment (nicktrav) [inline modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache]: One notable change in version 3.x is that a client cannot be modified after it has been created. The alternative is to create a _new_ client by converting back to a builder, modifying the relevant fields and the instantiating a new client. I've adopted this pattern throughout, where client mutations need to take place.

File: samples/client/petstore-security-test/java/okhttp-gson/build.gradle [hunk 3]
```diff
@@ -94,10 +94,11 @@ if(hasProperty('target') && target == 'android') {
...
 
 dependencies {
-    compile 'io.swagger:swagger-annotations:1.5.15'
-    compile 'com.squareup.okhttp:okhttp:2.7.5'
-    compile 'com.squareup.okhttp:logging-interceptor:2.7.5'
+    compile 'io.swagger:swagger-annotations:1.5.17'
+    compile 'com.squareup.okhttp3:okhttp:3.14.2'
+    compile 'com.squareup.okhttp3:logging-interceptor:3.14.2'
     compile 'com.google.code.gson:gson:2.8.1'
+    compile 'io.gsonfire:gson-fire:1.8.0'
```
Comment (nicktrav) [inline samples/client/petstore-security-test/java/okhttp-gson/build.gradle]: There are some additional changes in `petstore-security-test` that don't seem related to the OkHttp 3.x change. I'm guessing something wasn't updated in a previous commit.

**Comment by autodidacticon** [comment]
You might try submitting this here:
https://github.com/OpenAPITools/openapi-generator

**Comment by kgrosvenor** [comment]
Could someone point out to me how you generate okhttp v3? I always get 2.7.5?
